### ISSUE TITLE
Textures are not written correctly

### ISSUE THREAD

**Issue by Guekka** [Ryan-rsm-McKenzie/bsa#5]
I am able to reproduce this bug with this simple code
```cpp
#include <bsa/fo4.hpp>

int main()
{
    using namespace bsa::fo4;
    file f;
    f.read("textures/in.dds", format::directx);
    f.write("textures/out.dds", format::directx);
}
```
This corrupts all tested textures. : [Example](https://github.com/Ryan-rsm-McKenzie/bsa/files/7515079/Downloads.zip)

By looking at it in a hex editor, the header appears to be fine. The corruption starts at the end of the header, as soon as we enter the data section

**The issue appears to be in `write`** . Creating a textures BSA, and extracting it with another tool seems to work

**Comment by Guekka** [comment]
Well, I investigated a bit more. By looking more closely in a hex editor, the issue is simpler than I thought

Some (20) `CD` bytes were inserted right after the header and before the data. The rest of the data is fine, just shifted

**Comment by Guekka** [comment]
These bytes are in the place of the DX10 header. Maybe we need to force creation of a DX10 header

**Comment by Ryan-rsm-McKenzie** [comment]
fixed by #6
